pkgs/unified_analytics/lib/src/utils.dart

```diff
@@ -75,12 +77,88 @@ Directory getHomeDirectory(FileSystem fs) {
   } else if (io.Platform.isLinux) {
     home = envVars['HOME'];
   } else if (io.Platform.isWindows) {
-    home = envVars['UserProfile'];
+    home = envVars['AppData'];
   }
 
   return fs.directory(home!);
 }
 
+/// Returns `true` if user has opted out of legacy analytics in Dart or Flutter
+///
+/// Checks legacy opt-out status for the Flutter
+/// and Dart in the following locations
+///
+/// Dart: `$HOME/.dart/dartdev.json`
+///
+/// Flutter: `$HOME/.flutter`
+bool legacyOptOut({
+  required FileSystem fs,
+  required Directory home,
+}) {
+  final File dartLegacyConfigFile =
+      fs.file(p.join(home.path, '.dart', 'dartdev.json'));
+  final File flutterLegacyConfigFile = fs.file(p.join(home.path, '.flutter'));
+
+  // Example of what the file looks like for dart
+  //
+  // {
+  //   "firstRun": false,
+  //   "enabled": false,  <-- THIS USER HAS OPTED OUT
+  //   "disclosureShown": true,
+  //   "clientId": "52710e60-7c70-4335-b3a4-9d922630f12a"
+  // }
+  if (dartLegacyConfigFile.existsSync()) {
+    try {
+      // Read in the json object into a Map and check for
+      // the enabled key being set to false; this means the user
+      // has opted out of analytics for dart
+      final Map<String, Object?> dartObj =
+          jsonDecode(dartLegacyConfigFile.readAsStringSync());
+      if (dartObj.containsKey('enabled') && dartObj['enabled'] == false) {
+        return true;
+      }
+    } on FormatException {
+      // In the case of an error when parsing the json file, return true
+      // which will result in the user being opted out of unified_analytics
+      //
+      // A corrupted file could mean they opted out previously but for some
+      // reason, the file was written incorrectly
+      return true;
```

Comment:
I'm thinking we should delete the file if we couldn't parse it.

Reply:
> I'm thinking we should delete the file if we couldn't parse it.

I am against such drastic action; if there is a bug in how we are parsing the file we could interfere with the legacy analytics consent system.   Better to just leave it behind, as this is after all an optional feature.

Reply:
Eventually, once legacy analytics is removed, we can add something to delete legacy analytics files if we see them.